### ISSUE TITLE
The table block footer section is not displaying properly.

### ISSUE THREAD

**Issue by viralsampat-multidots** [WordPress/gutenberg#46183]
## What problem does this address?
I have checked Gutenberg table block and I have added its header section, test data, and footer section. But, in admin and at front-end user can't identify that which is footer section. Because footer section is react like row & columns data.

For better understanding here I have attached video for the same.
[Video](https://share.cleanshot.com/MnpeFOVomRFsY2DmakzI) 

## What is your proposed solution?
Here, I have one suggestion like if possible then we can modified footer section like our header section, If we will display like our header section. So user can easily identify that here we have footer section.

**Comment by devanshijoshi9** [comment]
Hey @t-hamano 👋 

To display the footer section the same as the header section, I have added the same style as `thead > th` tag in the [theme. scss](https://github.com/WordPress/gutenberg/blob/bea1bcf49421a2e52b7029868bcc316e37abaab4/packages/block-library/src/table/theme.scss#L8) file locally. 

Here is the output for the same: 

In editor: 
![table-block-1](https://user-images.githubusercontent.com/86941664/205561597-2f9209fc-89fb-40d5-98f3-767eeee2e649.png)

In front-end:
![table-block-2](https://user-images.githubusercontent.com/86941664/205561914-e73eea42-004b-4e56-af13-6a698cca38c5.png)

Let me know if it is the right implementation. Thanks.

**Comment by t-hamano** [comment]
@viralsampat-multidots 
Thank you for your contribution.

I saw your video and the borders below the `thead` and above the `tfoot` are defined in [theme.css](https://github.com/WordPress/gutenberg/blob/b333d1d515c7fe91ec063112a6c178d5ed66b1e4/packages/block-library/src/table/theme.scss). In other words, the borders are displayed in themes that have opted for block styles through `add_theme_support( 'wp-block-styles')`, as in Twenty Twenty Two.

In your video, this border disappears on the front end, but the latest Gutenberg trunk appears to have resolved this issue. This should not be a problem in themes that support block styles, since `thead` and `tfoot` can be identified by the border.

However, in themes without block styles like TT3, there remains the problem of not being able to _visually_ identify where the footer is.

![tfoot](https://user-images.githubusercontent.com/54422211/206827108-abd27bbb-71a9-4f0c-8316-f6ff9eca0693.png)

To solve this problem, @devanshijoshi9 tried the approach of making the `tfoot` font thicker, but the style change affects backward compatibility. In my opinion, we might refrain from changing the style for visual purposes, since the placeholder text will tell us that it is a footer before we enter text.

@mikachan @aaronrobertshaw @alexstine 
If you have any comments on this proposal or a better approach to solving this problem, I would love to hear your advice 🙏

**Comment by alexstine** [comment]
@joedolson Maybe some visual feedback you can provide?

**Comment by aaronrobertshaw** [comment]
> If you have any comments on this proposal

Thanks for the ping 👍 

I gave this a bit of a test and witnessed the same behaviour, and came to the [same conclusions](https://github.com/WordPress/gutenberg/issues/46183#issuecomment-1345038431) as @t-hamano.

> To solve this problem, @devanshijoshi9 tried the approach of making the tfoot font thicker, but the style change affects backward compatibility

To play devil's advocate for a moment, it could also be possible to view the footer's lack of distinction as a visual bug or accessibility issue. Depending on how subtle any styling tweaks were, they might be a welcome change.

That said, I don't have a strong opinion in this case, but I would likely lean towards not introducing anything potentially breaking as well.

There’s also been some long running discussion around [improving the Table block](https://github.com/WordPress/gutenberg/issues/32400), including [refactoring it to use inner blocks](https://github.com/WordPress/gutenberg/issues/18768) which would provide a huge increase in terms of what we could control and style. The downside there is any change on that front is going to be slow as its a huge and complex job.

**Comment by joedolson** [comment]
Overall, I'd like to see the table block refactored so that it offers a better overall experience. 

The biggest problem I can see with the lack of a visually distinct demarcation of `tfoot` is that users will simply treat it as an additional row of the table body, not noticing that it's actually a footer. Because the only visible marker is a placeholder, once you've added content to the cell, there's no obvious way to know that it's any different from other rows. 

That may be obvious when you create the block, but not necessarily to a second editor who comes in to make changes, later. 

I'd consider the lack of distinction to be a bug.

**Comment by t-hamano** [comment]
Thanks for your input, @aaronrobertshaw, @joedolson.

For the moment, I have added this issue as a bug in [the tracking issue](https://github.com/WordPress/gutenberg/issues/32400). (possibly an enhancement).

Also, @viralsampat-multidots has submitted #46567, which seems to offer a solution that includes making the font-weight thicker.

**Comment by ntsekouras** [comment]
Can we close this one since https://github.com/WordPress/gutenberg/pull/46567 has landed?

**Comment by t-hamano** [comment]
@ntsekouras 
Thanks for letting me know.

In #46567, `thead` and `tfoot` borders were finally moved from `theme.css` to `style.css`. So the borders are now visible regardless of whether or not wp-block-styles opts in.

I would like to close this issue, but if I have missed something please comment.

Twenty Twenty Three:

![tt3](https://user-images.githubusercontent.com/54422211/212226088-31c94584-41ff-4d45-b8d0-2a1ed91a4511.png)

Twenty Twenty One:

![tt1](https://user-images.githubusercontent.com/54422211/212226081-262cd753-7a8d-4fab-a9ea-528ec3acb41c.png)
